packages/travelmakers-design-core/src/components/Calendar/Calendar.tsx

```diff
@@ -0,0 +1,305 @@
+import { PolymorphicRef } from "@travelmakers-design-v2/styles";
+import React, {
+  PropsWithChildren,
+  forwardRef,
+  useEffect,
+  useState,
+} from "react";
+import { addDays, differenceInDays, getMonth, isEqual } from "date-fns";
+import { View } from "../View";
+import useStyles from "./Calendar.style";
+import {
+  CalendarProps,
+  CalendarComponent,
+  SelectedDays,
+} from "./Calendar.type";
+import {
+  useCalendar,
+  useToast,
+  useUpdateEffect,
+} from "@travelmakers-design-v2/hooks";
+import { DateCell } from "./_components/DateCell";
+import {
+  DateCellDay,
+  DateCellType,
+} from "./_components/DateCell/DateCell.type";
+import TableHead from "./_components/TableHead";
+import Indicator from "./_components/Indicator";
+import OptionBox from "./_components/OptionBox";
+import _ from "lodash";
+
+export interface Props {
+  /** 캘린더의 타입 */
+  type: "tour" | "move-in";
+
+  /** 선택 가능한 일자 */
+  selectableDates: Date[];
+
+  /** 선택가능한 최소 일자 */
+  minNight: number;
+
+  /** 선택가능한 최대 일자 */
+  maxNight: number;
+
+  onChange: (selected: SelectedDays) => void;
+
+  /** initial Date */
+  selected?: SelectedDays;
+
+  /** 선택 불가능한 일자 */
+  disabledDays?: Date[];
+
+  /** 허용하지 않는 날짜를 클릭하였을 경우 출력할 메시지 */
+  notAllowedMessage?: string;
+}
+
+/**
+ * ANCHOR: Travelmakers Calendar
+ * - type이 'tour(투어)' 일 경우 일자는 범위 선택이 아니라 하나의 일자에 대해서만 선택이 가능하다.
+ * - type이 'move-in(입주)' 일 경우 범위(from~to)형태로 일자를 선택한다.
+ * - selectableDates에 선택 가능한 일자를 넣어야만 정상적으로 일자에 대한 선택이 가능하다.
+ * - minNight로 최소 선택가능 범위를 작성해주어야한다.
+ * - maxNight로 최대 선택가능 범위를 작성해주어야한다.
+ * - onChange함수를 통해 선택한 일자에 대해서 확인할 수 있다.
+ */
+export const Calendar: CalendarComponent & {
+  displayName?: string;
+  OptionBox?: typeof OptionBox;
+} = forwardRef(
+  <C extends React.ElementType = "div">(
+    {
+      type = "move-in",
+      selected,
+      disabledDays = [],
+      selectableDates = [],
+      notAllowedMessage = "해당 일자는 예약이 불가합니다. \n다른 일자를 선택해주세요.",
+      minNight = 30,
+      maxNight = 59,
+      onChange,
+      children,
+      className,
+      ...props
+    }: PropsWithChildren<CalendarProps<C>>,
+    ref: PolymorphicRef<C>
+  ) => {
+    const { toast } = useToast();
+    const { classes, cx } = useStyles();
+    const [state, actions] = useCalendar(null);
+    const [enabledDays, setEnabledDays] = useState<Date>();
+    const [checked, setChecked] = useState({
+      from: selected?.from,
+      to: selected?.to,
+    });
+
+    useUpdateEffect(() => {
+      onChange?.(checked);
+    }, [checked]);
+
+    /**
+     * ANCHOR: 선택하지 않은 날짜 사이에 대해서 체크
+     * @param day
+     * @returns
+     */
+    const isBetweenNotSelectedDays = (day: DateCellDay) => {
+      if (
+        checked.from &&
+        !checked.to &&
+        differenceInDays(day.date, checked.from.date) > 0 &&
+        differenceInDays(enabledDays, day.date) >= 0
+      ) {
+        return true;
+      }
+      return false;
+    };
```

Comment:
P3; 유틸 성향이 강한 boolean 함수들은 if문을 제거하는게 어떨까요? 오히려 복잡한 느낌입니다.

```suggestion
    const isBetweenNotSelectedDays = (day: DateCellDay) => {
     return (
        checked.from &&
        !checked.to &&
        differenceInDays(day.date, checked.from.date) > 0 &&
        differenceInDays(enabledDays, day.date) >= 0
     )
    };
```

Reply:
✅